### ISSUE TITLE
[8.0.x] Maybe2 needs work

### ISSUE THREAD

**Issue by NeQuissimus** [scalaz/scalaz#1538]
I tried using Maybe2.
I understand the benefit of it not creating a Tuple2 instance.
However, since it does not have any type class instances other than `Show` and `IsCovariant` I am not sure how useful it is.
I assume that for the moment, it is a bit of a stub?

- `import scalaz.data.Maybe2._` is needed to create instances, the prelude is not sufficient
- I can't figure out how to access the `Show` instance :)
- without `Monoid`, `Monad` and friends, `Maybe2` is a little "bare"

I may try enhancing `Maybe2` myself, this issue is for tracking purposes mostly.

**Comment by TomasMikula** [comment]
Yeah, `Maybe2` is pretty barebones at the moment. It was introduced to support efficient implementation of `IList` via `Fix`. No bells and whistles were added.
